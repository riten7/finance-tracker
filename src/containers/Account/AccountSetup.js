import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useHistory } from 'react-router-dom';
import { Button, Divider, Header } from 'semantic-ui-react';

import AccountForm from '../../containers/Account/AccountForm';
import AccountList from './AccountList';
import { addAccount } from '../../actions/actionCreators';

import './account.css';

const AccountSetup = () => {
  const dispatch = useDispatch();
  const history = useHistory();
  
  const { accounts } = useSelector(state => state.accounts);

  const handleFormSubmit = (account) => {
    dispatch(addAccount(account));
  }

  const finalizeFormSubmit = () => {
    history.push("/dashboard");
  }

  return (
    <div className="account-setup-container">
        <Header as="h2" icon="settings" content="Ux-Cam Money Tracker" />
        <Divider />
        <Header as="h2">Accounts</Header>
        <p>
          Create accounts that you would like to keep track of.
          <br />
          It could be cash in your wallet, bank accounts, credit cards or even a
          loan to your friend.
        </p>
        <AccountForm handleFormSubmit={handleFormSubmit} />
        {accounts.length > 0 && (
          <div style={{ margin: '1em' }}>
            <AccountList />
            <div className="form-submit">
              <Button
                primary
                content="Finish"
                onClick={finalizeFormSubmit}
              />
            </div>
          </div>
        )}
      </div>
  )
}

export default AccountSetup;