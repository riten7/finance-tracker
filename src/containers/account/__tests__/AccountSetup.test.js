import React from 'react';
import { mount } from "enzyme";
import { Provider } from 'react-redux';
import configureStore from "redux-mock-store";
import AccountSetup from '../AccountSetup';
import data from '../../../__mockData__/data.json';

describe("AccountForm component", () => {
  let wrapper;
  const finalizeFormSubmit = jest.fn();
  const store = configureStore()(data);
   it("Component renders properly", () => {
    wrapper = mount(<Provider store={store}><AccountSetup /></Provider>);
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("Save button works properly", () => {
    //console.log(wrapper.find('.form .button').props())
    //wrapper.find('.form .form-submit').props().onClick();
    //expect(finalizeFormSubmit).toHaveBeenCalled();
  });
});