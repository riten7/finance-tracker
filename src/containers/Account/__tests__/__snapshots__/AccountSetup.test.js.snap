// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountForm component Component renders properly 1`] = `"<div class=\\"account-setup-container\\"><h2 class=\\"ui header\\"><i aria-hidden=\\"true\\" class=\\"settings icon\\"></i><div class=\\"content\\">Ux-Cam Money Tracker</div></h2><div class=\\"ui divider\\"></div><h2 class=\\"ui header\\">Accounts</h2><p><strong>Create accounts that you would like to keep track of. You can create maximum of 4 accounts that could be cash in your wallet, bank accounts, credit cards or even a loan to your friend.</strong></p><form class=\\"ui form account-form\\"><div class=\\"fields\\"><div class=\\"required eight wide field\\"><label>Name</label><div class=\\"ui input\\"><input name=\\"name\\" placeholder=\\"Account name\\" required=\\"\\" type=\\"text\\" value=\\"\\"></div></div><div class=\\"eight wide field\\"><label>Type</label><div name=\\"type\\" role=\\"listbox\\" aria-expanded=\\"false\\" class=\\"ui selection dropdown\\" tabindex=\\"0\\"><div aria-atomic=\\"true\\" aria-live=\\"polite\\" role=\\"alert\\" class=\\"divider text\\">Cash</div><i aria-hidden=\\"true\\" class=\\"dropdown icon\\"></i><div class=\\"menu transition\\"><div style=\\"pointer-events: all;\\" role=\\"option\\" aria-checked=\\"true\\" aria-selected=\\"true\\" class=\\"active selected item\\"><span class=\\"text\\">Cash</span></div><div style=\\"pointer-events: all;\\" role=\\"option\\" aria-checked=\\"false\\" aria-selected=\\"false\\" class=\\"item\\"><span class=\\"text\\">Bank Account</span></div><div style=\\"pointer-events: all;\\" role=\\"option\\" aria-checked=\\"false\\" aria-selected=\\"false\\" class=\\"item\\"><span class=\\"text\\">Credit</span></div><div style=\\"pointer-events: all;\\" role=\\"option\\" aria-checked=\\"false\\" aria-selected=\\"false\\" class=\\"item\\"><span class=\\"text\\">Deposit</span></div><div style=\\"pointer-events: all;\\" role=\\"option\\" aria-checked=\\"false\\" aria-selected=\\"false\\" class=\\"item\\"><span class=\\"text\\">Asset</span></div></div></div></div></div><div class=\\"eight wide field\\"><label>Balance</label><div class=\\"ui input\\"><input placeholder=\\"Balance\\" name=\\"balance\\" type=\\"number\\" value=\\"\\"></div></div><div class=\\"field\\"><button class=\\"ui primary button\\">Save Account</button></div></form><div style=\\"margin: 1em;\\"><h2>Accounts</h2><div class=\\"ant-collapse ant-collapse-icon-position-left\\"><div class=\\"ant-collapse-item\\"><div class=\\"ant-collapse-header\\" role=\\"button\\" tabindex=\\"0\\" aria-expanded=\\"false\\"><span role=\\"img\\" aria-label=\\"right\\" class=\\"anticon anticon-right ant-collapse-arrow\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" class=\\"\\" data-icon=\\"right\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z\\"></path></svg></span>Cash<div class=\\"ant-collapse-extra\\"><span class=\\"ant-typography ant-typography-success\\">NPR. 131406817</span></div></div></div></div><div class=\\"form-submit\\"><button class=\\"ui primary button\\">Finish</button></div></div></div>"`;
